### ISSUE TITLE
Add zfs_receive_corrupt_data module option

### ISSUE THREAD

**Issue by AndCycle** [zfsonlinux/zfs#3015]
this is just the other end that needs to be done if someone need to construct bad send stream,

we have zfs_send_corrupt_data module option, that can pump out data for rescue,
but the other end should also be able to feed with those bad data, or there is no one can receive that,

https://groups.google.com/a/zfsonlinux.org/forum/#!topic/zfs-discuss/uVWxFozSEcY
there is a recent discussion about someone trapped in a condition needs this.

http://thr3ads.net/zfs-discuss/2011/06/583545-ZFS-receive-checksum-mismatch
and an aged discussion mentioned where to modify.

**Comment by thegreatgazoo** [comment]
If I haven't missed something, I don't think zfs_receive_corrupt_data is needed. The 1st discussion was actually started by a customer of ours who I helped to recover their corrupted pool by using zfs_send_corrupt_data.

When zfs_send_corrupt_data is enabled, the sender just replaces corrupted data with a well-known pattern, and at the receive end the send stream would look perfectly good. There's a checksum that covers the whole send stream which is calculated from the well-known pattern. There's also checksums for individual WRITE records which are again calculated from the data with corrupted part replaced already.

In short, my experience and my understanding of code is that zfs_send_corrupt_data would work without any additional code needed on the receive side.
